src/libsystemd/sd-device/device-private.c

```diff
@@ -468,6 +468,12 @@ int device_new_from_nulstr(sd_device **ret, uint8_t *nulstr, size_t len) {
 
                 i += end - key + 1;
 
+                /* netlink messages for some devices contain an unwanted newline at the end of value.
+                 * Let's drop the newline and remaining strings after the newline if exists. */
```

Comment:
Please drop "if exists" — it's grammatically incorrect.

We have `truncate_nl` that does the same thing.